src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PushCommand.cs

```diff
@@ -105,7 +110,9 @@ public static void Register(CommandLineApplication app, Func<ILogger> getLogger)
                     }
 
 #pragma warning disable CS0618 // Type or member is obsolete
-                    var sourceProvider = new PackageSourceProvider(XPlatUtility.GetSettingsForCurrentWorkingDirectory(), enablePackageSourcesChangedEvent: false);
+                    var sourceProvider = new PackageSourceProvider(
+                        XPlatUtility.ProcessConfigFile(configurationFile.Value()),
```

Comment:
I think we should add a validation to ensure file path exists before accessing it.